libs/shared/flagd-core/src/lib/feature-flag.ts

```diff
@@ -18,12 +19,18 @@ export class FeatureFlag {
   private readonly _defaultVariant: string;
   private readonly _variants: Map<string, FlagValue>;
   private readonly _targeting: unknown;
+  private readonly _hash: string;
 
   constructor(flag: Flag) {
     this._state = flag['state'];
     this._defaultVariant = flag['defaultVariant'];
     this._variants = new Map<string, FlagValue>(Object.entries(flag['variants']));
     this._targeting = flag['targeting'];
+    this._hash = createHash('sha1').update(JSON.stringify(flag)).digest('base64');
```

Comment:
I _think_ this may be a bit less deterministic than you hope.

I think the hash would be different based on the order of the props in the `Flag` you pass in. This might not be that important, _but_, you could also generate this has based on a aggregation of the internal state, which I think would be more deterministic:

Something like: 

```ts
  this._hash = createHash('sha1').update(JSON.stringify(this)).digest('base64');
```

Reply:
This should be deterministic enough. We would emit a change event if someone modified the order of the properties in the object but that seems like an edge case and something we can easily explain.